src/decide.js

```diff
@@ -5,23 +5,18 @@ const logger = require("./logger");
 const { partial } = require("./utils");
 
 function compareFileTrees(a, b) {
-	if (a.length !== b.length) return false;
-	const sorter = (m, n) => (m.path < n.path ? -1 : m.path > n.path ? 1 : 0);
-	const sortedA = a.slice().sort(sorter);
-	const sortedB = b.slice().sort(sorter);
-
 	const cmp = (elOfA, elOfB) => {
 		const pathsAreEqual = elOfB.path === elOfA.path;
 		const lengthsAreEqual = elOfB.length === elOfA.length;
```

Comment:
I'm wondering now if we should do the same fuzzy checking on these file lengths. I realize it was like that before this PR, but since you're in the mood to weed out all the false negatives, this might be one of the sources of false negatives. Could you figure out if the `length` properties on individual files in multi-file torrents vary with piece size / align to piece size boundaries? If so then maybe we'll want to do something like a sizeDoesMatch call on them.

Reply:
I wouldn't touch the file size (length) checking, because It would fail to detect any tampering with the file. Do you mean path lengths? Perhaps it could be taken out, but I worry comparing just the file sizes is not enough. I don't know about the piece size you ask for, but when I did cross-seed manually the chunk size didn't matter for the end result so I wouldn't rely on that (i had a torrent with 1MB chunk and 16MB chunk and it cross-seeded just fine).

Reply:
Good point, let's keep that as-is then.